src/main/java/uk/gov/hmcts/reform/judicialapi/elinks/repository/ProfileRepository.java

```diff
@@ -24,7 +24,7 @@ public interface ProfileRepository extends JpaRepository<UserProfile, String> {
         + "LEFT JOIN FETCH judicialOfficeAppointment appt "
         + "on per.personalCode = appt.personalCode "
         + "LEFT JOIN FETCH judicialOfficeAuthorisation auth "
-        + "on auth.appointmentId = appt.appointmentId "
+        + "on auth.personalCode = appt.personalCode "
```

Comment:
I think this should:
* always match on personcode 👍 
* match on  `auth.appointmentId is null` or `auth.appointmentId = appt.appointmentId`

i.e. if there is an appointment ID then we should filter the authorisations JOIN on it: with personcode being the fallback for theose authorisations that have no parent appointment so should apply to all of the person's appointments.